segment/app/routes/contact.tsx

```diff
@@ -0,0 +1,36 @@
+import { json } from "@remix-run/node";
+import { Form } from "@remix-run/react";
+import { type SyntheticEvent } from "react";
+
+import * as segment from "~/utils/segment.client";
+
+export const action = async () => json({});
+
+export default function Contact() {
+  const handleSubmit = (e: SyntheticEvent<HTMLFormElement>) => {
+    const target = e.target as typeof e.target & {
+      message: { value: string };
+    };
+
+    console.log(`message: '${target.message.value}'`);
+
+    // The event is using `track` to keep it as simple as posible.
+    // It could use `trackForm` instead. See: https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#track-form
+    segment.track("Form Submitted", {
+      message: target.message.value,
+    });
+  };
```

Comment:
yes, doing it in the action would require also a segment server-side implementation, and while it is possible, requires some additional workarounds to sync the server-side and client-side anonymous ids, user ids, and some logic to prevent duplicated events in destinations (e.g. Google Analytics). Actually, that way is how I implemented it for work, but I thought that would add too much complexity to this example, but lmk

Reply:
@MichaelDeBoey Good, I'll see if I have some time next week to work on it